memory/Makefile

```diff
@@ -0,0 +1,42 @@
+include ../config.mk
+
+PYTHONPATH := $${PYTHONPATH:+$${PYTHONPATH}:}../hail/python:../gear
+PYTHON := PYTHONPATH=$(PYTHONPATH) python3
+
+MEMORY_LATEST = gcr.io/$(PROJECT)/memory:latest
+MEMORY_IMAGE = gcr.io/$(PROJECT)/memory:$(shell docker images -q --no-trunc memory | sed -e 's,[^:]*:,,')
+
+ifeq ($(NAMESPACE), "default")
+BATCH_PODS_NS = batch-pods
+else
+BATCH_PODS_NS = $(NAMESPACE)
+endif
```

Comment:
This is basically what Dan does for the shuffler---it errors in deploy if `NAMESPACE` isn't set.

I think he wants to move away from using makefiles for deploying stuff (which I think we already don't do for most stuff), so the only real use case I see for the deploy step in the Makefile is to deploy things for testing in my dev namespace without having to deploy all the dependencies, as I would need to do using dev deploy. I don't actually want to deploy to the default namespace unless I'm really actively definitely trying to replace the thing in production, which I don't think is going to be the common use case, if any; I certainly don't want to accidentally omit the namespace and deploy something that I was trying to test.

The actual thing I want to do here is to just rely on the user to have their kubeconfig set correctly, with the correct namespace, but I don't think anyone else uses that; thoughts?

Reply:
OK, fair.  Since shuffler is already doing it, let's address this separately.

Yeah, we're all agreement about the Makefiles.  We should build a local tool that executes build.yaml.

 > the only real use case I see for the deploy step in the Makefile is to deploy things for testing in my dev namespace without having to deploy all the dependencies

Ah, so I had a rough plan to address this, but it never got executed.  There were a few variations I haven't decided among.  One issues is there are really two kinds of dependencies: coordination between external state (e.g. deploy of a service you depend on) and "value" dependencies that produce configuration data used by the step (e.g. build_image is used by deploy_service in the k8s yaml).  You either have to run the value dependencies, or take them from the current values deployed in default (I think that's always possible but I haven't checked).  So basically I wanted an option for `dev deploy` like `--no-deps` or `--assume a,b,c` that skipped a,b,c (and their dependencies) if the requested steps depend on them.  FYI @danking

Reply:
> I don't actually want to deploy to the default namespace unless I'm really actively definitely trying to replace the thing in production, which I don't think is going to be the common use case, if any; I certainly don't want to accidentally omit the namespace and deploy something that I was trying to test.

 > The actual thing I want to do here is to just rely on the user to have their kubeconfig set correctly, with the correct namespace, but I don't think anyone else uses that; thoughts?

I agree, the Makefile here is slightly different than the others.  The others are used to bootstrap a fresh cluster (or manually patch a critical issue).  Nobody uses them for test deployment (I think default is hardwired in most of them).

I do set the kubectl namespace a lot.  I have this alias:

```
kcd () {
    k config set-context ... --namespace $1
}
```

This would be reasonable to use.  There is still some danger deploying to default.